### PR TITLE
Fix plumbing channels with caps & try to fix flaky tests again

### LINKED ISSUE #237: Cannot plumb channels with a capital letter
If a channel is created on Freenode with `#SomeCaps`, it can't be provisioned because the bridge never seems to be able to join the room. I suspect this is due to case mapping rules going awry (e.g. it listens for `#SomeCaps` but the reply comes back as `#somecaps` and so things it never got the JOIN callback). We need to case map asap when we get a provisioning request according to ISUPPORT rules (retrieved from MatrixBridge)

### PR DESCRIPTION
**Pull request by lukebarnard1**
Case map channels as per ISUPPORT on /[query]link, and add test to check that channels are case mapped during provisioning.

Attempts to fix #237

### CODE REVIEW

File: lib/provisioning/Provisioner.js [hunk 1]
```diff
@@ -609,6 +609,15 @@ Provisioner.prototype.handlePm = Promise.coroutine(function*(server, fromUser, t
     );
 });
 
+// Using ISUPPORT rules supported by MatrixBridge bot, case map ircChannel
+function caseFold(cli, channel) {
+    if (!cli.unsafeClient) {
+        log.warn(`Could not case map ${channel} - BrdigedClient has no IRC client`);
```
Comment (Kegsay) [inline lib/provisioning/Provisioner.js]: `BrdigedClient`

**Comment by Kegsay** [comment]
Otherwise LGTM (as per IRL, do the `afterEach` to fix poor ol' Travis)

**Comment by lukebarnard1** [comment]
Looks like flakiness is still plaguing us, because the `beforeEach` of one test may not be executed the `afterEach` of the next? Both provide the `done()` argument, so using `test.coroutine` in `afterEach` should be fine.
```
2016-10-19 09:54:01 INFO:main Killing bridge
2016-10-19 09:54:01 INFO:IrcBridge Killing all clients
2016-10-19 09:54:01 INFO:BridgedClient Killing client  ro_bot_nick
2016-10-19 09:54:01 INFO:client-connection disconnect()ing ro_bot_nick@irc.example - Bridged client killed
IrcClient.emit => ["irc.example_ro_bot_nick","disconnect",{"addr":"irc.example","nick":"ro_bot_nick","opts":{"userName":"matrixirc","realName":"matrixirc","autoConnect":false,"autoRejoin":false,"floodProtection":true,"floodProtectionDelay":700,"port":6667,"selfSigned":false,"retryCount":0,"family":4,"bustRfc3484":true},"chans":{},"_events":{"error":[null,null],"message":[null,null],"notice":[null,null],"ctcp-privmsg":[null,null]},"_eventsCount":19},"Bridged client killed",null]
2016-10-19 09:54:01 INFO:BridgedClient Killing client  ro_bot_nick
2016-10-19 09:54:01 INFO:BridgedClient Killing client  ro_bot_nick
link endpoint : should not create a M<--->I link when remote_room_server is not defined
=======================================================================================
Add listener(irc.example_M-flibble) for fn=connect
Add listener(irc.example_bob) for fn=connect
Add listener(irc.example_ro_bot_nick) for fn=connect
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_bob) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=names
Add listener(irc.example_ro_bot_nick) for fn=part
2016-10-19 09:54:02 ERROR:IrcBridge Failed to join channel:: #coffee
2016-10-19 09:54:02 ERROR:IrcBridge 
2016-10-19 09:54:06 ERROR:BridgedClient Could not try to join: no client for M-alice, channel = #coffee
```

**Comment by lukebarnard1** [comment]
IRL'd and discussed that ENOENT is causing trouble again, and expanding the above sample downwards, we see:
```
2016-10-19 09:54:01 INFO:main Killing bridge
2016-10-19 09:54:01 INFO:IrcBridge Killing all clients
2016-10-19 09:54:01 INFO:BridgedClient Killing client  ro_bot_nick
2016-10-19 09:54:01 INFO:client-connection disconnect()ing ro_bot_nick@irc.example - Bridged client killed
IrcClient.emit => ["irc.example_ro_bot_nick","disconnect",{"addr":"irc.example","nick":"ro_bot_nick","opts":{"userName":"matrixirc","realName":"matrixirc","autoConnect":false,"autoRejoin":false,"floodProtection":true,"floodProtectionDelay":700,"port":6667,"selfSigned":false,"retryCount":0,"family":4,"bustRfc3484":true},"chans":{},"_events":{"error":[null,null],"message":[null,null],"notice":[null,null],"ctcp-privmsg":[null,null]},"_eventsCount":19},"Bridged client killed",null]
2016-10-19 09:54:01 INFO:BridgedClient Killing client  ro_bot_nick
2016-10-19 09:54:01 INFO:BridgedClient Killing client  ro_bot_nick
link endpoint : should not create a M<--->I link when remote_room_server is not defined
=======================================================================================
Add listener(irc.example_M-flibble) for fn=connect
Add listener(irc.example_bob) for fn=connect
Add listener(irc.example_ro_bot_nick) for fn=connect
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_bob) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=names
Add listener(irc.example_ro_bot_nick) for fn=part
2016-10-19 09:54:02 ERROR:IrcBridge Failed to join channel:: #coffee
2016-10-19 09:54:02 ERROR:IrcBridge 
2016-10-19 09:54:06 ERROR:BridgedClient Could not try to join: no client for M-alice, channel = #coffee
link endpoint : should not create a M<--->I link when remote_room_channel is not defined
========================================================================================
2016-10-19 09:54:07 ERROR:BridgedClient Could not try to join: no client for M-alice, channel = #coffee
Add listener(irc.example_M-flibble) for fn=connect
Add listener(irc.example_bob) for fn=connect
Add listener(irc.example_ro_bot_nick) for fn=connect
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_bob) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=names
Add listener(irc.example_ro_bot_nick) for fn=part
2016-10-19 09:54:08 ERROR:BridgedClient Could not try to join: no client for ro_bot_nick, channel = #provisionedchannel
FATAL
Error: ENOENT: no such file or directory, rename 'spec-db/users.db~' -> 'spec-db/users.db'
    at Error (native)
2016-10-19 09:54:09 DEBUG:DataStore removeConfigMappings 
2016-10-19 09:54:09 INFO:DataStore Indexes checked on 'id' for store.
2016-10-19 09:54:09 INFO:DataStore Indexes checked on 'matrix_id' for store.
2016-10-19 09:54:09 INFO:DataStore Indexes checked on 'remote_id' for store.
```

File: lib/main.js [hunk 2]
```diff
@@ -130,6 +133,13 @@ module.exports.runBridge = Promise.coroutine(function*(port, config, reg) {
 
     // run the bridge
     ircBridge = new IrcBridge(config, reg);
+
+    // Force paths to stores to be undefined so that Nedb uses in memory DB instead
```
Comment (Kegsay) [inline lib/main.js]: Lying comment :)

**Comment by Kegsay** [comment]
LGTM aside from a single lying comment.

**Pull request merged**